hugo/content/4-updates/operators.md

```diff
@@ -0,0 +1,367 @@
+
+    // Decrement the time left in the free trial of some product.
+    // The following three statements are the same, and produce the BSON:
+    // { $inc: {trial_days_left: -1}}
+    auto res = collection.update_many(MANGROVE_KEY(User::is_using_trial) == true,
+                                      MANGROVE_KEY(User::trial_days_left) -= 1);
+    res = collection.update_many(MANGROVE_KEY(User::is_using_trial) == true,
+                                 MANGROVE_KEY(User::trial_days_left)--);
+    res = collection.update_many(MANGROVE_KEY(User::is_using_trial) == true,
+                                 --MANGROVE_KEY(User::trial_days_left));
+    ```
+
+- `{{% a_blank "$mul" "https://docs.mongodb.com/manual/reference/operator/update/mul/" %}}`
+
+    The `$mul` operator is provided as the C++ `*=` operator.
+    It is only enabled for fields with a numeric type.
+
+    ```cpp
+    // Double the score of a user who has a bonus in some game.
+    // { $mul: {score: 2}}
+    auto res = collection.update_many(MANGROVE_KEY(User::has_bonus) == true, MANGROVE_KEY(User::score) *= 2);
+    ```
+
+- `{{% a_blank "$rename" "https://docs.mongodb.com/manual/reference/operator/update/rename/" %}}`
+
+    *Mangrove does not provide the `$rename` operator, since changing the name of fields creating during serialization is not possible.*
+
+- `{{% a_blank "$setOnInsert" "https://docs.mongodb.com/manual/reference/operator/update/setOnInsert/" %}}`
+
+    This operator is provided as a member function on the name-value pairs.
+    The given value must be the same type as the field (or implicitly cast to it).
+
+    ```cpp
+    // Reset the score of a user, or create a new one.
+    // Note the use of the "upsert" option, to enable insertion of a new document if necessary.
+    // { $set: {score: 0}, $setOnInsert: {hours_played: 0} }
+    auto res = collection.update_one(MANGROVE_KEY(User::username) == "raphofkhan",
+                                     (MANGROVE_KEY(User::score) = 0,
+                                      MANGROVE_KEY(User::hours_played).set_on_insert(0)),
+                                     mongocxx::options::update{}.upsert(true));
+    ```
+
+- `{{% a_blank "$set" "https://docs.mongodb.com/manual/reference/operator/update/set/" %}}`
+
+    The `$set` operator is provided as the C++ assignment operator `=`.
+    The given value must be the same type as the field (or implicitly cast to it).
+
+    ```cpp
+    // Edit the username of a user
+    // { $set: {username: "brohemian_raphsody"}}
+    auto res = collection.update_one(MANGROVE_KEY(User::username) == "raphofkhan",
+                                    MANGROVE_KEY(User::username) = "brohemian_raphsody");
+    ```
+
+- `{{% a_blank "$unset" "https://docs.mongodb.com/manual/reference/operator/update/unset/" %}}`
+
+    This operator is provided as a member function on name-value pairs that takes no arguments.
+    The field it is used on must be an `optional` field.
```

Comment:
Is it a compile-time error or runtime error if not used on an optional? If it's run-time, add a warning.